src/libcore/tests/slice.rs

```diff
@@ -1127,12 +1127,13 @@ fn sort_at_index() {
                 // Sort in ascending order.
                 for pivot in 0..len {
                     let mut v = orig.clone();
-                    v.sort_at_index_by(pivot, |a, b| a.cmp(b));
+                    let (left, pivot, right) = v.partition_at_index_by(pivot, |a, b| a.cmp(b));
 
-                    assert_eq!(v_sorted[pivot], v[pivot]);
-                    for i in 0..pivot {
-                        for j in pivot..len {
-                            assert!(v[i] <= v[j]);
+                    for l in left.iter_mut() {
+                        assert!(l <= pivot);
+                        for r in right.iter_mut() {
```

Comment:
Since the `.iter_mut()` is redundant, the style can be `for r in right {`

Reply:
This gives the error:

```
value moved here in previous iteration of loop
```

for `right`.

Reply:
I figured it out. :)

Reply:
I think you have a good point, it needs .iter_mut() or reborrow in the inner loop. Uh, I can't see how `&mut left = mutable slice` should compile

Reply:
Sorry -- I must have gotten something cached during the build phase. Hope this is better?

Reply:
Thanks